### DISCUSSION TITLE
CodeReady Linux Builder repository mandatory

### DISCUSSION THREAD

**Discussion by bdouxx** [glpi-project/glpi-agent#401] [Q&A]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

None

### Describe the bug

in the redhat documentation (red_hat_enterprise_linux 8) it's write:
`The CodeReady Linux Builder repository contains additional packages for use by developers. This repository is available with all Red Hat Enterprise Linux subscriptions but does not need to be enabled on your runtime deployments. Packages included in the CodeReady Linux Builder repository are unsupported`

we don't want to enable this repository in all servers in production environment for security reasons.
Do you really need dependencies on this repository?

### To reproduce

```
sudo yum install glpi-agent
Error:
 Problem: conflicting requests
  - nothing provides perl(Data::UUID) needed by glpi-agent-1.4-3.el8.noarch
  - nothing provides perl(DateTime) needed by glpi-agent-1.4-3.el8.noarch
  - nothing provides perl(HTTP::Daemon) needed by glpi-agent-1.4-3.el8.noarch
  - nothing provides perl(YAML::Tiny) needed by glpi-agent-1.4-3.el8.noarch
```

### Expected behavior

installation successfull without CodeReady Linux Builder repository

### Operating system

Linux

### GLPI Agent version

1.4

### GLPI version

10.0.6

**Comment by g-bougard** [comment]
Hello @bdouxx 

the short answer is GLPI Agent needs few perl libraries which are only provided by that repository when installing by rpm on this distro.

Now you have few options:
1. Enable temporarily that repository during glpi-agent installation and disable it afterward
2. Don't enable the repository but download required packages manually and install them manually before glpi-agent
3. Use snap version of the package
4. Use AppImage package

Honestly, the redhat documentation is lying about the fact "The CodeReady Linux Builder repository contains additional packages for use by developers.". This is indeed what they wanted for that repository, but the reality is different for perl libraries. They can be used by softwares and not only during development. The sentence "Packages included in the CodeReady Linux Builder repository are unsupported" only means Redhat won't answer on any support request related to the provided packages. The libraries the agent uses are fully open-source and so are supported by the community. About security, I don't know for other provided packages, but the one the agent uses are really safe as far I know. To me the "security" reason is a wrong reason here. Be sure if any provided package could be a security problem redhat would remove it immediately from the repository.
